crates/validation/src/lib.rs

```diff
@@ -105,9 +82,63 @@ pub async fn validate(
     }
     storage_mapping::walk_all_storage_mappings(storage_mappings, &mut errors);
 
+    // Names of collection which use inferred schemas.
+    let inferred_collections = reference::gather_inferred_collections(collections);
+    // Names of collections which are referenced, but are not being validated themselves.
+    let remote_collections =
+        reference::gather_referenced_collections(captures, collections, materializations, tests);
+
+    // Concurrently fetch referenced collections and inferred schemas from the control-plane.
+    let (inferred_schemas, remote_collections) = match futures::try_join!(
+        control_plane.get_inferred_schemas(inferred_collections),
+        control_plane.resolve_collections(remote_collections),
+        // TODO(johnny): Also fetch storage mappings here.
+    ) {
+        Ok(ok) => ok,
+        Err(err) => {
+            // If we failed to fetch from the control-plane then further validation
+            // will generate lots of misleading errors, so fail now.
+            Error::ControlPlane { detail: err }.push(root_scope, &mut errors);
+            return tables::Validations {
+                built_captures: tables::BuiltCaptures::new(),
+                built_collections: tables::BuiltCollections::new(),
+                built_materializations: tables::BuiltMaterializations::new(),
+                built_tests: tables::BuiltTests::new(),
+                errors,
+            };
+        }
+    };
+
+    let remote_collections = remote_collections
+        .into_iter()
+        .map(|mut spec| {
+            tracing::debug!(collection=%spec.name, "resolved referenced remote collection");
+
+            // Clear a derivation (if there is one), as we do not need it
+            // when embedding a referenced collection.
+            spec.derivation = None;
```

Comment:
This should be `derive` now. I checked, and there's no longer any `live_specs` that use `derivation`, so I think we're good to remove it altogether once we get around to it.

Reply:
somehow this comment got duplicated